### PR TITLE
#19452: Contramap and fromFunction

### PR DESCRIPTION
**Pull request by drewhk**
refs: #19452

### CODE REVIEW

**Comment by drewhk** [comment]
credit goes to @aloiscochard

**Comment by ktoso** [comment]
You could `git commit --amend --author 'Alois Cochard <alois.cochard@gmail.com>'` :)

File: akka-stream-tests/src/test/java/akka/stream/javadsl/FlowTest.java [hunk 1]
```diff
@@ -803,6 +803,17 @@ public Integer create() {
     assertEquals((Object) 0, result);
   }
 
+  @Test
+  public void shouldBePossibleToCreateFromFunction() throws Exception {
+    //Source(0 to 9).via(Flow.fromFunction(_ + 1)).runWith(Sink.seq).futureValue should ===(1 to 10)
```
Comment (viktorklang) [inline akka-stream-tests/src/test/java/akka/stream/javadsl/FlowTest.java]: Why?

**Comment by aloiscochard** [comment]
@drewhk @ktoso no worries for the credits folks!

Thanks a lot for taking this where I left it, much appreciated :-)

File: akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala [hunk 2]
```diff
@@ -285,6 +285,11 @@ object Flow {
   def apply[T]: Flow[T, T, NotUsed] = identity.asInstanceOf[Flow[T, T, NotUsed]]
 
   /**
+   * Creates a `Flow` from a function.
```
Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala]: That doc is a tad unconvincing, mate :)

File: akka-stream-tests/src/test/java/akka/stream/javadsl/FlowTest.java (hunk 1 above)
Comment (drewhk) [inline akka-stream-tests/src/test/java/akka/stream/javadsl/FlowTest.java]: ah, forgot to remove.

File: akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala (hunk 2 above)
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala]: If you have any idea what to add, please share and I will add it :)

**Comment by drewhk** [comment]
now the credit is all yours

File: akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala (hunk 2 above)
Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala]: `Creates a `Flow` which will use the given function `f` to transform its inputs to outputs.`

**Comment by 2m** [comment]
LGTM

**Comment by johanandren** [comment]
Unrelated failure (#19741)

**Comment by rkuhn** [comment]
LGTM

**Pull request merged**